google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -1264,6 +1266,71 @@ public Database getDatabase(String databaseName) throws SpannerException {
         NanoClock.getDefaultClock());
   }
 
+
+  @Override
+  public OperationFuture<Backup, CopyBackupMetadata> copyBackUp(
+          final com.google.cloud.spanner.Backup backupInfo) throws SpannerException {
+    final String instanceName = backupInfo.getInstanceId().getName();
+    final String databaseName = backupInfo.getDatabase().getName();
+    final String backupId = backupInfo.getId().getBackup();
+    final Backup.Builder backupBuilder =
+            com.google.spanner.admin.database.v1.Backup.newBuilder()
+                    .setDatabase(databaseName)
+                    .setExpireTime(backupInfo.getExpireTime().toProto());
+    if (backupInfo.getVersionTime() != null) {
+      backupBuilder.setVersionTime(backupInfo.getVersionTime().toProto());
+    }
+
+    final CopyBackupRequest.Builder requestBuilder =
+            CopyBackupRequest.newBuilder()
+                    .setParent(instanceName)
+                    .setBackupId(backupId)
+                    .setSourceBackup(backupId)
+                    .setExpireTime(backupInfo.getExpireTime().toProto()) ;
+
+    if (backupInfo.getEncryptionConfig() != null) {
+      requestBuilder.setEncryptionConfig(
+              EncryptionConfigProtoMapper.copyBackupEncryptionConfig(
+                      backupInfo.getEncryptionConfig()));
+    }
+    final CopyBackupRequest request = requestBuilder.build();
+    final OperationFutureCallable<CopyBackupRequest, Backup, CopyBackupMetadata> callable =
+            new OperationFutureCallable<>(
+                    databaseAdminStub.copyBackupOperationCallable(),
+                    request,
+                    //calling copy backup method of dbClientImpl
+                    DatabaseAdminGrpc.getCopyBackupMethod(),
+                    instanceName,
+                    nextPageToken ->
+                            listBackupOperations(
+                                    instanceName,
+                                    0,
+                                    String.format(
+                                            "(metadata.@type:type.googleapis.com/%s) AND (metadata.name:%s)",
+                                            CopyBackupMetadata.getDescriptor().getFullName(),
+                                            String.format("%s/backups/%s", instanceName, backupId)),
+                                    nextPageToken),
+                    input -> {
+                      try {
+                        return input
+                                .getMetadata()
+                                .unpack(CopyBackupMetadata.class)
+                                .getProgress()
+                                .getStartTime();
+                      } catch (InvalidProtocolBufferException e) {
+                        return null;
```

Comment:
returning null makes me worried. Should we throw a RuntimeException?

Reply:
I think this is OK in this case. We use this to order operations by start time, and this just ensures that if the `listBackupOperations` happens to return an operation without a start time (or some other invalid operation) that we just ignore it.